core/modules/loader/CentralWorkerDoListItem.h

```diff
@@ -0,0 +1,67 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ *
+ */
+
+#ifndef LSST_QSERV_LOADER_CENTRALWORKERDOLISTITEM_H
+#define LSST_QSERV_LOADER_CENTRALWORKERDOLISTITEM_H
+
+// Qserv headers
+#include "loader/CentralWorker.h"
+
+namespace lsst {
+namespace qserv {
+namespace loader {
+
+/// This class exists to regularly call the CentralWorker::_monitor() function, which
+/// does things like monitor TCP connections and control shifting with the right neighbor.
+class CentralWorkerDoListItem : public DoListItem {
+public:
+    CentralWorkerDoListItem() = delete;
+    explicit CentralWorkerDoListItem(CentralWorker* centralWorker) : _centralWorker(centralWorker) {
+        setTimeOut(std::chrono::seconds(7));
+    }
+
+    util::CommandTracked::Ptr createCommand() override {
+        struct CWMonitorCmd : public util::CommandTracked {
+            CWMonitorCmd(CentralWorker* centralW) : centralWorker(centralW) {}
+            void action(util::CmdData*) override {
+                centralWorker->_monitor();
+            }
+            CentralWorker* centralWorker;
+        };
+        util::CommandTracked::Ptr cmd(new CWMonitorCmd(_centralWorker));
```

Comment:
prefer make_shared

Reply:
I've been using new because if I change the constructor parameters, I get nice simple compile errors instead of massive template issues. It's always been my intention to hunt them down and switch them once the code is stable. I'm not sure it is there yet.